libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RoomListExtensions.kt

```diff
@@ -40,9 +41,15 @@ fun RoomList.loadingStateFlow(): Flow<RoomListLoadingState> =
                 trySendBlocking(state)
             }
         }
-        val result = loadingState(listener)
-        send(result.state)
-        result.stateStream
+        tryOrNull {
+            val result = loadingState(listener)
+            try {
```

Comment:
It's the opposite, I don't think it's good practice to catch `Throwable`, as it's usually errors from the JVM and not the responsability of the app to handle them... I don't think `tryOrNull` should catch them neither, but thats another story.
Here, `loadingState` can fail if the `RoomList` has been destroyed async, which can happen when navigating quickly.
`onInitialList` can also fail if the scope is cancelled. As we need to return the `TaskHandle` it was the easiest way of doing so. Otherwise I can just create a var and have a unique try/catch, I don't mind, if you think it's more clear.

Reply:
I think so, let's try `var` + single `try/catch` then :) thanks!